File: FigurasGeometricas/EntidadesVb/TanqueRectangular.vb
Public Class TanqueRectangular
	Inherits Figura
	Implements Figura3D

	Public Property Base As Single
	Public Property Altura As Single
	Public Property Profundidad As Single
	Public Sub New(nombre As String, base As Single, altura As Single, profundidad As Single)
		MyBase.New(nombre)
		Me.Base = base
		Me.Altura = altura
		Me.Profundidad = profundidad
	End Sub

	Public Function CalcularVolumen() As Single Implements Figura3D.CalcularVolumen
		Return Base * Altura * Profundidad
	End Function
	Public Overrides Function ToString() As String
		Return $"Soy {Nombre} y tengo la altura de {Altura}, la base de {Base} y la profundidad de {Profundidad}"
	End Function
End Class

File: FigurasGeometricas/EntidadesCs/TanqueRectangular.cs
namespace EntidadesCs
{
	public class TanqueRectangular : Figura, Figura3D
	{
		public float Altura { get; set; }
		public float Base { get; set; }
		public float Profundidad { get; set; }
		public TanqueRectangular(string nombre, float @base, float altura, float profundidad) : base(nombre)
		{
			Base = @base;
			Altura = altura;
			Profundidad = profundidad;
		}
		public float CalcularVolumen()
		{
			return Base * Altura * Profundidad;
		}
		public override string ToString()
		{
			return $"Soy {Nombre} y tengo la base de {Base}, la altura de {Altura} y la profundidad de {Profundidad}";
		}

	}
}
